app/helpers/publishers_helper.rb

```diff
@@ -33,64 +41,77 @@ def publisher_humanize_balance(publisher, currency)
     I18n.t("helpers.publisher.balance_unavailable")
   end
 
-  def next_deposit_date(today = DateTime.now)
-    if today.day > 8
-      today = today + 1.month
+  def publisher_converted_overall_balance(publisher)
+    return if publisher.default_currency == "BAT" || publisher.default_currency.blank?
+    if overall_balance = publisher.wallet &&
+       publisher.wallet.overall_balance.is_a?(Eyeshade::OverallBalance) &&
+       publisher.wallet.overall_balance.amount_default_currency.present? &&
+       publisher.wallet.overall_balance
+       '%.2f' % overall_balance.amount_default_currency
+
+      I18n.t("helpers.publisher.balance_pending_approximate",
+             amount: '%.2f' % overall_balance.amount_default_currency,
+             code: overall_balance.default_currency)
+    else
+      I18n.t("helpers.publisher.conversion_unavailable", code: publisher.default_currency)
     end
-    today.strftime("%B 8th")
   end
 
-  def publisher_converted_balance(publisher)
-    currency = publisher.default_currency
-    return if currency == "BAT" || currency.blank?
-    if balance = publisher.wallet &&
-        publisher.wallet.contribution_balance.is_a?(Eyeshade::Balance) &&
-        publisher.wallet.contribution_balance
-      converted_amount = '%.2f' % balance.convert_to(currency)
-      I18n.t("helpers.publisher.balance_pending_approximate", amount: converted_amount, code: currency)
+  def publisher_channel_bat_balance(publisher, channel_identifier)
+    if balance = (
+        publisher.wallet &&
+        publisher.wallet.channel_balances &&
+        publisher.wallet.channel_balances[channel_identifier]
+    )
+
+      '%.2f' % balance.amount_bat
     else
-      I18n.t("helpers.publisher.conversion_unavailable", code: currency)
+      I18n.t("helpers.publisher.balance_unavailable")
     end
   rescue => e
     require "sentry-raven"
```

Comment:
I've pretty much checked with sentry and didn't find any errors in this helper function.Should we keep these exception trackers in?

Reply:
So I think rescuing the error here is important in the case that there is some unexpected state, or if there are issues with what eyeshade returns.  Since we're rescuing here, I think we might as well report the error to sentry.

I'm generally trying to be cautious until these changes have been stable in production for awhile.